### DISCUSSION TITLE
[Question] Behavior of DV_VALUE with Hicks and Henne bump functions

### DISCUSSION THREAD

**Discussion by r-ashwin** [su2code/SU2#1255] [Q&A]
If I set DV_KIND = HICKS_HENNE, then what exactly does DV_VALUE do? I see that it deforms the mesh, but I am trying to understand how exactly DV_VALUE (a scalar) is related to the vector of design variables defined in DEFINITION_DV?

Thanks for your help and apologies if this is documented elsewhere--I searched for it a bit but could not find a definition.

**Edit1**:
Just to add some context, I want to write a Python function that implements `f, df = function(x)`, where `x` is the vector of design variables (e.g., Hicks and Henne bump function coefficients). I tried to modify the `shape_optimization.py` to implement this, where I essentially ignore the calls to the optimizer (and I use `project.obj_f(x)` and `project.obj_df(x)` for this). However, the time it takes to evaluate the objective function and gradient this way, suggest that it might not be utilizing parallel computing (note that the code is successfully compiled and tested for parallel scale up). On the other hand, running `discrete_adjoint.py` for the same config file, computes the forward and adjoint solutions at a fraction of the time. For example, with 36 cores, the first approach takes ~15minutes whereas the second approach takes ~1minute. 

Therefore when I run `discrete_adjoint.py`, the script reads the mesh file to infer the geometry deformations and runs the forward and adjoint solvers on it. It appears that the way to generate a deformed mesh is to adjust `DV_VALUE` in the config file, run `SU2_DEF` and then used the deformed mesh. However, I am not sure how the scalar-valued `DV_VALUE` is related to the design vector `x`?

**Edit2**:
It appears that `DV_VALUE` essentially does the deformation at `DV_PARAM`.

How can I pass an array of `DV_VALUE` at list of `DV_PARAM` tuples? When I just tried passing a comma-separated list, I see that it ignores all inputs but the first one.

![su2_airfoils](https://user-images.githubusercontent.com/11855653/113932590-6722a600-97b9-11eb-962f-3d7fa8d3e424.PNG)

**Comment by r-ashwin** [comment]
@pcarruscag Would appreciate some feedback when you get a chance. Thank you!

**Reply by pcarruscag** [reply]
I never used them, hopefully some other SU2 volunteer will chime in on this one.
Cheers
PS Attach the config.

**Reply by r-ashwin** [reply]
Okay thanks feel free to tag the most appropriate volunteer.

I have attached the config file. I set `DV_PARAM = (1, 0.25), (0, 0.25)` and `DV_VALUE = 0.001, 0.001`. It ignores the second `()` in `DV_PARAM`.

[turb_SA_RAE2822.cfg.txt](https://github.com/su2code/SU2/files/6289111/turb_SA_RAE2822.cfg.txt)

**Reply by pcarruscag** [reply]
For FFD you need as many DV_KIND as DV_PARAM and DV_VALUE, might be the same for HH

**Reply by r-ashwin** [reply]
Turns out that is true. Thanks!
